magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merging_runs.py

```diff
@@ -0,0 +1,571 @@
+
+Only MC:
+$ python merging_runs_and_splitting_training_samples.py --analysis-type onlyMC (-c config.yaml)
+"""
+
+import argparse
+import glob
+import logging
+import os
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+from tqdm import tqdm
+
+from magicctapipe import __version__
+
+__all__ = ["cleaning", "split_train_test", "merge", "mergeMC"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def cleaning(list_of_nodes, cwd):
+
+    """
+    This function looks for failed runs in each node and remove them.
+
+    Parameters
+    ----------
+    list_of_nodes : array of str
+        List of nodes where the function will look for failed runs.
+    cwd : Path
+        Current working directory
+    """
+
+    for i in tqdm(range(len(list_of_nodes)), desc="Cleaning failed runs"):
+        os.chdir(list_of_nodes[i])
+        os.system('find . -type f -name "*.h5" -size -1k -delete')
+
+    os.chdir(cwd)
+    print("Cleaning done.")
+
+
+def split_train_test(target_dir, train_fraction, source_name):
+
+    """
+    This function splits the MC proton sample in 2, i.e. the "test" and the "train" subsamples.
+    It generates 2 subdirectories in the directory .../DL1/MC/protons named "test" and "train" and creates sub-sub-directories with the names of all nodes.
+    For each node sub-sub-directory we move 80% of the .h5 files (if it is in the "test" subdirectory) or 20% of the .h5 files (if it is in the "train" subdirectory).
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    train_fraction : float
+        Fraction of proton MC files to be used in the training RF dataset
+    source_name : str
+        Name of the target source
+    """
+
+    proton_dir = f"{target_dir}/{source_name}/DL1/MC/protons"
+
+    if not os.path.exists(f"{proton_dir}/train"):
+        os.mkdir(f"{proton_dir}/train")
+    if not os.path.exists(f"{proton_dir}/../protons_test"):
+        os.mkdir(f"{proton_dir}/../protons_test")
+
+    list_of_dir = np.sort(glob.glob(f"{proton_dir}/node*{os.path.sep}"))
+
+    for directory in tqdm(
+        range(len(list_of_dir))
+    ):  # tqdm allows us to print a progessbar in the terminal
+        if not os.path.exists(
+            f"{proton_dir}/train/{list_of_dir[directory].split('/')[-2]}"
+        ):
+            os.mkdir(f"{proton_dir}/train/{list_of_dir[directory].split('/')[-2]}")
+        if not os.path.exists(
+            f"{proton_dir}/../protons_test/{list_of_dir[directory].split('/')[-2]}"
+        ):
+            os.mkdir(
+                f'{proton_dir}/../protons_test/{list_of_dir[directory].split("/")[-2]}'
+            )
+        list_of_runs = np.sort(
+            glob.glob(f'{proton_dir}/{list_of_dir[directory].split("/")[-2]}/*.h5')
+        )
+        split_percent = int(len(list_of_runs) * train_fraction)
```

Comment:
actually it is not a percent, but a simple number (number of runs to be moved), so the name of the variable is a bit misleading